src/coreclr/tests/src/Interop/UnmanagedCallersOnly/UnmanagedCallersOnlyDll.cpp

```diff
@@ -36,20 +36,33 @@ namespace
         int Result;
     };
 
-    void ProxyCall(ProxyCallContext* cxt)
+    void* ProxyCall(void* cxtRaw)
     {
+        auto cxt = (ProxyCallContext*)cxtRaw;
         cxt->Result = CallManagedProc(cxt->CallbackProc, cxt->N);
+        return NULL;
     }
 }
 
 extern "C" DLL_EXPORT int STDMETHODCALLTYPE CallManagedProcOnNewThread(CALLBACKPROC pCallbackProc, int n)
 {
     ProxyCallContext cxt{ pCallbackProc, n, 0 };
+
+#ifdef _WIN32
     std::thread newThreadToRuntime{ ProxyCall, &cxt };
 
     // Wait for new thread to complete
     newThreadToRuntime.join();
 
+#else // _WIN32
+    pthread_t newThreadToRuntime;
+    int ret = pthread_create(&newThreadToRuntime, NULL, ProxyCall, &cxt);
```

Comment:
Do you receive the return value that you don't use just to avoid linter warnings?